### Scenario
A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.
### Data
A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29
### Assignment Tasks
a. Build 99% Confidence Interval Using Sample Standard Deviation
Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.

b. Build 99% Confidence Interval Using Known Population Standard Deviation
If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.


### Import required libraries

##### a. Build 99% Confidence Interval Using Sample Standard Deviation Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.

In [4]:
# Scipy for Stats Function Data and Pandas and Numpy for Data Manupulation
from scipy import stats
import math
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### Get Sample Data

In [5]:
df = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

In [6]:
df = pd.DataFrame(df)
df

,0
0,1.13
1,1.55
2,1.43
3,0.92
4,1.25
5,1.36
6,1.32
7,0.85
8,1.07
9,1.48


In [7]:
df.rename({0:'Sample'}, axis=1, inplace=True)

In [8]:
df['Sample'].mean()

1.2386666666666666

In [9]:
df['Sample'].std()

0.19316412956959936

In [10]:
df.describe()

,Sample
count,15.000000
mean,1.238667
std,0.193164
min,0.850000
25%,1.155000
50%,1.250000
75%,1.345000
max,1.550000


In [12]:
# Confidnce Interval = (n, loc/Mean, SD/Scale)

sample_mean = df['Sample'].mean()
sample_std_dev = df['Sample'].std()
sample_size = df['Sample'].count()
confidence_level = 0.99

## Degrees of freedom
dof = sample_size - 1

## Significance Level
alpha = (1 - confidence_level) / 2

## finding t- value from t distibution table
t_value = stats.t.ppf(1 - alpha, dof)

margin_of_error = t_value * (sample_std_dev / np.sqrt(sample_size))

lower_limit = sample_mean - margin_of_error
upper_limit = sample_mean + margin_of_error

print("dof:", dof)
print("alpha:", alpha)
print("t_value:", t_value)
print("margin_of_error:", margin_of_error)

print(f"99% Confidence Interval for the mean number of characters printed before the print-head fails using sample Standard Deviation :")
print(f"Confidence Interval : ({lower_limit}, {upper_limit})")

dof: 14
alpha: 0.0050000000000000044
t_value: 2.976842734370834
margin_of_error: 0.14846932822817596
99% Confidence Interval for the mean number of characters printed before the print-head fails using sample Standard Deviation :
Confidence Interval : (1.0901973384384906, 1.3871359948948425)


### Observtion :

99% confidence interval ranges between (1.0901973384384906, 1.3871359948948425) for the mean number of characters printed before the print-head fails using the sample standard deviation.
Using t test, cosidering the sample as population and the standard deviation is not known.

##### b. Build 99% Confidence Interval Using Known Population Standard Deviation If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.

In [ ]:
## confidence Interval = xbar +- z * (sigma / sqrt(n))

## left_limit = xbar - Z * (sigma / sqrt(n))
## right_limit = xbar  Z * (sigma / sqrt(n))

Z value for 99% confidence level from Z table

In [16]:
xbar = df['Sample'].mean()
Z = 2.58
pop_std_dev = 0.2
n = len(df['Sample'])

left_limit = xbar - ( Z * (pop_std_dev / np.sqrt(n)))
right_limit = xbar + ( Z * (pop_std_dev / np.sqrt(n)))

print(f"99% confidence interval for the mean number of characters printed before failure for known population standard deviation 0.2 is :")
print(f"Confidence Interval : ({left_limit}, {right_limit})")

99% confidence interval for the mean number of characters printed before failure for known population standard deviation 0.2 is :
Confidence Interval : (1.1054360395571314, 1.3718972937762017)


### Observation : 
The confidence interval with 99% with known standard deviation is (1.1054360395571314, 1.3718972937762017)

### Conclusion :
#### We can conclude that the confidence interval with 99%, with unknown population standard deviation using t test and with known population standard deviation at 0.2 using Z test is approximatetly equal.